include/deal.II/particles/particle_handler.h

```diff
@@ -79,8 +79,9 @@ namespace Particles
     /**
      * A type for the storage container for particles.
      */
-    using particle_container =
-      std::vector<std::vector<typename PropertyPool<dim, spacedim>::Handle>>;
+    using particle_container = std::list<
+      std::pair<std::vector<typename PropertyPool<dim, spacedim>::Handle>,
+                typename Triangulation<dim, spacedim>::active_cell_iterator>>;
```

Comment:
I wonder if we can define a `struct ParticlesInCell` that contains a `std::vector<typename PropertyPool<dim, spacedim>::Handle> particles;` and a `typename Triangulation<dim, spacedim>::active_cell_iterator> cell_iterator`.
 Then a lot of the calls below would be easier to read. Instead of `particles_in_cell->first/second` it would be `particles_in_cell->particles/cell_iterator`. Do you think that would have a performance impact?

Reply:
That's a good idea and helps documenting all first/second instances. It won't have any performance impact. I will change in the next few days.

Reply:
@gassmoeller this is now in f7c21cd